kernel/src/fs/cgroupfs/systree_node.rs

```diff
@@ -542,3 +697,66 @@ where
 
     op(process, &mut cgroup_guard)
 }
+
+enum SubControlAction {
+    Activate(String),
+    Deactivate(String),
+}
+
+/// Reads the actions for sub-control from the given reader.
+///
+/// Returns a tuple containing vector of actions and the number of bytes read.
+fn read_subtree_control_from_reader(
+    reader: &mut VmReader,
+) -> Result<(Vec<SubControlAction>, usize)> {
```

Comment:
There are no needs to return `Vec` here. Return two `SubCtrlState` flags: one denotes the controllers that need to be deactivated and another one denotes the controllers that need to be activated.